pkg/operator/ceph/object/user/controller.go

```
@@ -0,0 +1,229 @@
+	var ok bool
+	objectstoreuser, ok = obj.(*cephv1beta1.ObjectStoreUser)
+	if ok {
+		// the objectstoreuser object is of the latest type, simply return it
+		return objectstoreuser.DeepCopy(), nil
+	}
+	return nil, fmt.Errorf("not a known objectstoreuser object: %+v", obj)
+}
+
+// Create the user
+func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cephv1beta1.ObjectStoreUser) error {
+	// validate the user settings
+	if err := ValidateUser(context, u); err != nil {
+		return fmt.Errorf("invalid user %s arguments. %+v", u.Name, err)
+	}
+
+	// create the user
+	logger.Infof("creating user %s in namespace %s", u.Name, u.Namespace)
+	userConfig := cephrgw.ObjectUser{
+		UserID:      u.Name,
+		DisplayName: &u.Name,
+	}
+	objContext := cephrgw.NewContext(context, u.Spec.Store, u.Namespace)
+
+	user, rgwerr, err := cephrgw.CreateUser(objContext, userConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create user %s. RadosGW returned error %d: %+v", u.Name, rgwerr, err)
+	}
+
+	// Store the keys in a secret
+	secrets := map[string]string{
+		"AccessKey": *user.AccessKey,
+		"SecretKey": *user.SecretKey,
+	}
+	secret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rook-ceph-object-user-" + u.Name,
+			Namespace: u.Namespace,
+		},
+		StringData: secrets,
+		Type:       k8sutil.RookType,
+	}
+	k8sutil.SetOwnerRef(context.Clientset, u.Namespace, &secret.ObjectMeta, &c.ownerRef)
+
+	_, err = context.Clientset.CoreV1().Secrets(u.Namespace).Create(secret)
+	if err != nil {
+		return fmt.Errorf("failed to save user %s secret. %+v", u.Name, err)
+	}
+	logger.Infof("created user %s", u.Name)
+	return nil
+}
+
+// Delete the user
+func deleteUser(context *clusterd.Context, u *cephv1beta1.ObjectStoreUser) error {
+	objContext := cephrgw.NewContext(context, u.Spec.Store, u.Namespace)
+	result, rgwerr, err := cephrgw.DeleteUser(objContext, u.Name)
+	if err != nil {
```

Can we differentiate if the user is not found? In that case we could ignore the error.